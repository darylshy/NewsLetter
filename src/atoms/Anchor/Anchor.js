import React from "react";
import PropTypes from "prop-types";
import classes from "./Anchor.module.scss";
// import { Link } from "react-router-dom";

const Anchor = props => {
  return (
    <a className={classes.Anchor} href="#">
      {props.text}
    </a>
  );
};

Anchor.propTypes = {
  text: PropTypes.string.isRequired
};

export default Anchor;
