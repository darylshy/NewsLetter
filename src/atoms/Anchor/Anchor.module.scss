@import "../../styles/global/globals.scss";
.Anchor {
  font-family: "Azo-Sans-Light";
  color: $font-grey;
  @include desktop {
    font-size: 1.2rem;
  }
  @include tabletL {
    font-size: 1.2rem;
  }
  @include tabletP {
    font-size: 1.2rem;
  }
  @include mobile {
    font-size: 1.2rem;
  }
}
