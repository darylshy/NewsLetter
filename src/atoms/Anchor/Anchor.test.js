import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Anchor from './Anchor';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Anchor />', () => {
  it('should render an anchor tag', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Anchor link="Link" />
        </MemoryRouter>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
