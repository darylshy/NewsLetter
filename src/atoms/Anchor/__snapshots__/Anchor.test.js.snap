// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should render an anchor tag 1`] = `
<a
  className="Anchor"
  href="/Link"
  onClick={[Function]}
>
  Link
</a>
`;
